src/hotspot/share/runtime/crac.cpp

```diff
@@ -422,12 +424,23 @@ Handle crac::checkpoint(jarray fd_arr, jobjectArray obj_arr, bool dry_run, jlong
   Universe::heap()->set_cleanup_unused(false);
   Universe::heap()->finish_collection();
 
+  AsyncLogWriter* aio_writer = AsyncLogWriter::instance();
+  if (aio_writer) {
+    aio_writer->stop();
+  }
+  LogConfiguration::close();
+
   VM_Crac cr(fd_arr, obj_arr, dry_run, (bufferedStream*)jcmd_stream);
   {
     MutexLocker ml(Heap_lock);
     VMThread::execute(&cr);
   }
   if (cr.ok()) {
+    LogConfiguration::reopen();
```

Comment:
Looks like the log stays closed & blocked if the checkpoint fails?

Reply:
Hopefully it is fixed now?